csg/share/scripts/inverse/inverse.sh

```diff
@@ -182,12 +182,35 @@ steps_done=0
 i="$(( $begin - 1 ))"
 while true; do
   ((i++))
-  if [[ -z ${do_iterations} ]]; then
-    iterations_max="$(csg_get_property cg.inverse.iterations_max)"
+  # decide if to do step
+  iterations_max="$(csg_get_property --allow-empty cg.inverse.iterations_max)"
+  # first check if iterations_max and do_iterations are int
+  if [[ -n $iterations_max ]]; then
     is_int "$iterations_max" || die "inverse.sh: cg.inverse.iterations_max needs to be a number, but I got $iterations_max"
-    echo "We are doing $i of $iterations_max iterations (0=inf)."
-    [[ $iterations_max -ne 0 && $i -gt $iterations_max ]] && break
   fi
+  if [[ -n $do_iterations ]]; then
+    is_int "$do_iterations" || die "inverse.sh: option --do-iterations needs to be a number, but I got $do_iterations"
+  fi
+  # if --do-iterations was provided, decide by it
+  if [[ -n $do_iterations ]]; then
+    # exception: if iterations_max is zero, ignore do_iterations. Needed for some tests in csg_tutorial (i.e. spce/t-hncn/pre)
+    if [[ $iterations_max == 0 && $do_iterations -gt 0 ]]; then
+      msg "Stopping at step $i, user requested to do more, but iterations_max=0 indicates this should only run step000"
+      break
+    fi
+    # stop according to --do-iterations
+    if [[ $do_iterations -eq $steps_done ]] ; then
+      msg "Stopping at step $i, user requested to take some rest after this amount of iterations"
+      exit 0  # not creating done file
+    else
+      msg "Going on for another $(( $do_iterations - $steps_done )) steps"
+    fi
+   # stop according to iterations_max
+  else
+    [[ $iterations_max -ne -1 && $i -gt $iterations_max ]] && break
+    echo "We are doing $i of $iterations_max iterations (-1=inf, 0=only step_000)."
```

Comment:
This should be documented in `csg_defaults.xml` and maybe empty `iterations_max` should be `inf` instead of `-1`.

Reply:
Hmm or maybe `-1` is fine.

Reply:
> This should be documented in `csg_defaults.xml` and maybe empty `iterations_max` should be `inf` instead of `-1`.

It is now documented in `csg_defaults.xml`